hack/generator/pkg/codegen/pipeline_add_cross_resource_references.go

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package codegen
+
+import (
+	"context"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+	"k8s.io/klog/v2"
+
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
+	"github.com/Azure/azure-service-operator/hack/generator/pkg/config"
+)
+
+var armIDDescriptionRegex = regexp.MustCompile(".*/subscriptions/.*?/resourceGroups/.*")
+
+// TODO: For now not supporting array or map of references. Unsure if it actually ever happens in practice.
+
+// addCrossResourceReferences replaces cross resource references with genruntime.ResourceReference.
+func addCrossResourceReferences(configuration *config.Configuration, idFactory astmodel.IdentifierFactory) PipelineStage {
+	return MakePipelineStage(
+		"addCrossResourceReferences",
+		"Replaces cross resource references with genruntime.ResourceReference",
+		func(ctx context.Context, definitions astmodel.Types) (astmodel.Types, error) {
+
+			result := make(astmodel.Types)
+			knownReferences := newKnownReferencesMap(configuration)
+			isCrossResourceReference := func(typeName astmodel.TypeName, prop *astmodel.PropertyDefinition) bool {
+				ref := referencePair{
+					typeName: typeName,
+					propName: prop.PropertyName(),
+				}
+				_, isReference := knownReferences[ref]
```

Comment:
This is a hardcoded map, the heuristic is just for logging (for now). I think eventually the heuristic will leave our code and move into the Azure REST API Specs repo linter (at least, that's my plan), and the hardcoded list will be based on the Swagger specification annotations that the heuristic guides people to add.

Which is to say: I don't think this needs to be a map to `bool` because `knownReferences` is really just a set of hardcoded references - if something isn't a reference just don't include it in the set.